Validation/DTRecHits/plugins/DT2DSegmentClients.cc

```diff
@@ -1,43 +1,37 @@
-#include "DT2DSegmentClients.h"
-#include "FWCore/ServiceRegistry/interface/Service.h"
-#include "DQMServices/Core/interface/MonitorElement.h"
 #include "DQMServices/Core/interface/DQMStore.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
-
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
 #include "Validation/DTRecHits/interface/utils.h"
 
-//#include "TFile.h"
-#include <string>
-#include <iostream>
-#include <map>
-
+#include "DT2DSegmentClients.h"
 
 using namespace std;
 using namespace edm;
 
-DT2DSegmentClients::DT2DSegmentClients(const edm::ParameterSet& ps){
-}
-DT2DSegmentClients::~DT2DSegmentClients(){
+DT2DSegmentClients::DT2DSegmentClients(edm::ParameterSet const& config)
+{
 }
-void DT2DSegmentClients::endJob(){}
 
-void DT2DSegmentClients::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg,
-  edm::EventSetup const& c){
-  dbe = Service<DQMStore>().operator->();
-  //dbe->setCurrentFolder("/DT/2DSegments/");
-  Tutils * aux = new Tutils;
-  MonitorElement * hResPos = dbe->get("DT/2DSegments/Res/2D_SuperPhi_hResPos");
-  MonitorElement * hResAngle = dbe->get("DT/2DSegments/Res/2D_SuperPhi_hResAngle");
-  MonitorElement * hPullPos = dbe->get("DT/2DSegments/Pull/2D_SuperPhi_hPullPos");
+void DT2DSegmentClients::endLuminosityBlock(edm::LuminosityBlock const& lumi,
+  edm::EventSetup const& setup)
+{
+  DQMStore* dbe = Service<DQMStore>().operator->();
```

Comment:
DT2DSegmentClients should become DQMEDHarvester. DQMStore should really never be accessed.

The analyze is empty and harvester does gives you the dqmEndLuminosityBlock call.

Reply:
As I wrote 4 days ago
> this should take care of the migration to concurrent DQM; I did not try to address the part of the code that should be moved to the harvesting step.

The DT or DQM groups can take care of this further migration in a separate PR.